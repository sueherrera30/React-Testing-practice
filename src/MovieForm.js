import React, { Component } from 'react';

export default class MovieForm extends Component {
  state = {
    message: false,
  };

  showMessage = () => {
    this.setState(() => ({
      message: true,
    }));
  };

  render() {
    const { message } = this.state;
    return (
      <form data-testid="movie-form">
        <input type="text" data-testid="movie-input" />
        <button onClick={this.showMessage}>Submit</button>
        {
          message && (
          <p>
congratulations, form sent!
          </p>
          )
        }
      </form>
    );
  }
}
