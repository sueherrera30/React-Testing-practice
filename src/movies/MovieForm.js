import React, { Component } from 'react';

export default class MovieForm extends Component {
  state = {
    text: '',
  };

  render() {
    const { text } = this.state;
    const { sumbitForm } = this.props;
    return (
      <form
        data-testid="movie-form"
        onSubmit={() => sumbitForm({
          text,
        })}
      >
        <label htmlFor="text">
          Text
          <input
            type="text"
            data-testid="movie-input"
            id="text"
            onChange={event => this.setState({ text: event.target.value })}
          />
        </label>
        <button type="submit">
Submit
        </button>
      </form>
    );
  }
}

// export default class MovieForm extends Component {
//   state = {
//     message: false,
//   };

//   showMessage = () => {
//     this.setState(() => ({
//       message: true,
//     }));
//   };

//   render() {
//     const { message } = this.state;
//     return (
//       <form data-testid="movie-form">
//         <input type="text" data-testid="movie-input" />
//         <button onClick={this.showMessage}>Submit</button>
//         {
//           message && (
//           <p>
// congratulations, form sent!
//           </p>
//           )
//         }
//       </form>
//     );
//   }
// }
