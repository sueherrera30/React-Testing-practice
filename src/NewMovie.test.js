import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import MovieForm from './MovieForm';

afterEach(cleanup);

test('<MovieForm />', () => {
  const { getByTestId, getByText } = render(<MovieForm />);
  expect(getByTestId('movie-input').tagName).toBe('INPUT');
  const submitbutton = getByText('Submit');
  expect(submitbutton).toBeTruthy();
  fireEvent.click(submitbutton);
  expect(getByText('congratulations, form sent!').tagName).toBe('P');
});
