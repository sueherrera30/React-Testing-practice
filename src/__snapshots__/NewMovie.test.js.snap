// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie /> test snapshot 1`] = `
<div>
  <h1
    data-testid="page-title"
  >
    New Movie
  </h1>
  <form
    data-testid="movie-form"
  >
    <label
      for="text"
    >
      Text
      <input
        type="text"
        data-testid="movie-input"
        id="text"
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
